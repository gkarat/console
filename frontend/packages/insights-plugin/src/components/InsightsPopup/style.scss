.co-insights__no-rules {
  color: var(--pf-global--Color--200);
}

.insights-advisor--enabled-link {
  tspan {
    fill: var(
      --pf-global--link--Color
    ) !important; // use !important to override inline style applied by Victory charts

    &:hover {
      fill: var(--pf-global--link--Color--hover) !important;
    }
  }
}
