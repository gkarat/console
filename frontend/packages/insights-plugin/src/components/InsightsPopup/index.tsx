import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { PrometheusHealthPopupProps } from '@console/plugin-sdk';
import { K8sResourceKind } from '@console/internal/module/k8s';
import { ExternalLink, openshiftHelpBase } from '@console/internal/components/utils';

import AdvisorChart from '../AdvisorChart';
import {
  mapMetrics,
  isWaitingOrDisabled as _isWaitingOrDisabled,
  isError as _isError,
} from '../../mappers';
import './style.scss';

export const InsightsPopup: React.FC<PrometheusHealthPopupProps> = ({ responses, k8sResult }) => {
  const { t } = useTranslation();
  const metrics = mapMetrics(responses[0].response);
  const clusterId = (k8sResult as K8sResourceKind)?.data?.spec?.clusterId || '';
  const isWaitingOrDisabled = _isWaitingOrDisabled(metrics);
  const isError = _isError(metrics);

  return (
    <div className="co-insights__box">
      <p id="insights-plugin--header">
        {t(
          'insights-plugin~Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters.',
        )}
      </p>
      {isError && (
        <div className="co-status-popup__section text-secondary">
          {t('insights-plugin~Temporary unavailable.')}
        </div>
      )}
      {isWaitingOrDisabled && (
        <div className="co-status-popup__section text-secondary">
          {t('insights-plugin~Disabled or waiting for results.')}
        </div>
      )}
      {!isWaitingOrDisabled && !isError && <AdvisorChart metrics={metrics} clusterId={clusterId} />}
      <div className="co-status-popup__section">
        {!isWaitingOrDisabled && !isError && clusterId && (
          <>
            <h6 className="pf-c-title pf-m-md">{t('insights-plugin~Fixable issues')}</h6>
            <ExternalLink
              href={`https://cloud.redhat.com/openshift/details/${clusterId}#insights`}
              text={t('insights-plugin~View all in OpenShift Cluster Manager')}
            />
          </>
        )}
        {!isWaitingOrDisabled && !isError && !clusterId && (
          <ExternalLink
            href={`https://cloud.redhat.com/openshift/`}
            text={t('insights-plugin~Go to OpenShift Cluster Manager')}
          />
        )}
        {(isWaitingOrDisabled || isError) && (
          <ExternalLink
            href={`${openshiftHelpBase}support/remote_health_monitoring/using-insights-to-identify-issues-with-your-cluster.html`}
            text={t('insights-plugin~More about Insights')}
          />
        )}
      </div>
    </div>
  );
};

InsightsPopup.displayName = 'InsightsPopup';
