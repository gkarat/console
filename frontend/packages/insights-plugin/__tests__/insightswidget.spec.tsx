import * as React from 'react';
import { shallow } from 'enzyme';
import { InsightsPopup } from '../src/components/InsightsPopup/index';
import fixtures from './insights.fixtures';

describe('InsightsPopup ', () => {
  const props = {
    ...fixtures,
    responses: { ...fixtures.propsInsightsPopup.responses },
    k8sResult: { ...fixtures.propsInsightsPopup.k8sResult },
  };

  const wrapper = shallow(<InsightsPopup {...props} />);
  it('main container renders correctly', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should have the correct error component', () => {
    expect(wrapper.find('.co-insights__box').length).toBe(1);
  });

  it('should have correct text inside .co-insights__box', () => {
    expect(wrapper.find('.insights-widget-text-enabled').text()).toBe(
      'Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters',
    );
  });

  it('should render component co-status-popup__section', () => {
    expect(wrapper.find('.co-status-popup__section'));
  });
});

describe('InsightsPopup error state', () => {
  const { responses } = fixtures.propsInsightsPopup;
  responses[0].response.data.result = [];
  const props = {
    ...fixtures,
    responses,
  };
  const wrapper = shallow(<InsightsPopup {...props} />);

  it('renders the error component correctly', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
  it('should have the correct error component', () => {
    expect(wrapper.find('.co-status-popup__section-text-secondary').length).toBe(1);
  });
  it('should have correct text inside co-status-popup__section-text-secondary', () => {
    expect(wrapper.find('.insights-widget-text-error').text()).toBe('Temporary unavailable.');
  });
});

describe('InsightsPopup disabled state', () => {
  const props = {
    ...fixtures,
    responses: {
      ...fixtures.propsInsightsPopup.responses,
      response: {
        ...fixtures.propsInsightsPopup.responses[0].response,
        data: {
          ...fixtures.propsInsightsPopup.responses[0].response.data,
          result: {
            ...fixtures.propsInsightsPopup.responses[0].response.data.result,
            0: { value: [1622715672.165, '-1'] },
            1: { value: [1622715672.165, '-1'] },
            2: { value: [1622715672.165, '-1'] },
            3: { value: [1622715672.165, '-1'] },
          },
        },
      },
    },
    k8sResult: { ...fixtures.propsInsightsPopup.k8sResult },
  };
  const wrapper = shallow(<InsightsPopup {...props} />);

  it('renders the isWaitingOrDisabled component correctly', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
  it('should have the correct disabled component', () => {
    expect(wrapper.find('.co-status-popup__section-text-secondary').length).toBe(1);
  });
  it('should have correct text inside co-status-popup__section-text-secondary', () => {
    expect(wrapper.find('.insights-widget-text-disabled').text()).toBe(
      'Disabled or waiting for results.',
    );
  });
});
