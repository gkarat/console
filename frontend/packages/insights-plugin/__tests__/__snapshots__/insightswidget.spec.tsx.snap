// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InsightsPopup  main container renders correctly 1`] = `
"<div className=\\"co-insights__box\\">
  <p className=\\"insights-widget-text-enabled\\">
    Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters
  </p>
  <AdvisorChart metrics={{...}} clusterId=\\"\\" />
  <div className=\\"co-status-popup__section\\">
    <Component href=\\"https://cloud.redhat.com/openshift/\\" text=\\"Go to OpenShift Cluster Manager\\" />
  </div>
</div>"
`;

exports[`InsightsPopup disabled state renders the isWaitingOrDisabled component correctly 1`] = `
"<div className=\\"co-insights__box\\">
  <p className=\\"insights-widget-text-enabled\\">
    Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters
  </p>
  <div className=\\"co-status-popup__section-text-secondary\\">
    <p className=\\"insights-widget-text-disabled\\">
      Disabled or waiting for results.
    </p>
  </div>
  <div className=\\"co-status-popup__section\\">
    <Component href=\\"https://docs.okd.io/latest/support/remote_health_monitoring/using-insights-to-identify-issues-with-your-cluster.html\\" text=\\"More about Insights\\" />
  </div>
</div>"
`;

exports[`InsightsPopup error state renders the error component correctly 1`] = `
"<div className=\\"co-insights__box\\">
  <p className=\\"insights-widget-text-enabled\\">
    Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters
  </p>
  <div className=\\"co-status-popup__section-text-secondary\\">
    <p className=\\"insights-widget-text-error\\">
      Temporary unavailable.
    </p>
  </div>
  <div className=\\"co-status-popup__section\\">
    <Component href=\\"https://docs.okd.io/latest/support/remote_health_monitoring/using-insights-to-identify-issues-with-your-cluster.html\\" text=\\"More about Insights\\" />
  </div>
</div>"
`;
